src/sensor/ping.cpp

```diff
@@ -13,6 +13,46 @@
 #include <QThread>
 #include <QUrl>
 
+Ping::Ping() : Sensor() {
+    _points.reserve(_num_points);
+    for (int i = 0; i < _num_points; i++) {
+        _points.append(0);
+    }
+    _parser = new PingParser();
+    connect(dynamic_cast<PingParser*>(_parser), &PingParser::newMessage, this, &Ping::handleMessage);
+    connect(link(), &AbstractLink::newData, _parser, &Parser::parseBuffer);
+    emit linkUpdate();
+
+    _requestTimer.setInterval(1000);
+    connect(&_requestTimer, &QTimer::timeout, this, [this]{ request(PingMessage::es_profile); });
+
+    //connectLink("2:/dev/ttyUSB2:115200");
+
+    connect(&_detector, &ProtocolDetector::_detected, this, &Ping::connectLink);
+    _detector.start();
+
+    connect(this, &Ping::autoDetectUpdate, this, [this](bool autodetect){
+        if(!autodetect) {
+            if(_detector.isRunning()) {
+                _detector.exit();
+            }
+        } else {
+            if(!_detector.isRunning()) {
+                _detector.start();
+            }
+        }
+    });
+}
+
+void Ping::connectLink(const QString& connString)
+{
+    if(_detector.isRunning()) {
+        _detector.exit();
+    }
+    setAutoDetect(false);
```

Comment:
This line indicates that autodetect should not be allowed to be true at this point. What's the point of this line if the user can just click the checkbox to set it true it again?